### PR TITLE
Don't throw on deleteAttribute if column doesn't exist

### PR DESCRIPTION
**Pull request by PineappleIOnic**
This PR makes it so that when you deleteAttributes, if the column for the attribute does not exist, the adapter will still return true so the metadata can be cleaned up appropriately. Tests are not included for this as replicating the environment for this isn't easy with the tests architecture we have

### CODE REVIEW

File: src/Database/Adapter/MariaDB.php [hunk 1]
```diff
@@ -419,9 +419,17 @@ public function deleteAttribute(string $collection, string $id, bool $array = fa
 
         $sql = $this->trigger(Database::EVENT_ATTRIBUTE_DELETE, $sql);
 
-        return $this->getPDO()
+        try {
+            return $this->getPDO()
             ->prepare($sql)
             ->execute();
+        } catch (PDOException $e) {
```
Comment (abnegate) [inline src/Database/Adapter/MariaDB.php:426]: Indentation here is off, same for Postgres

File: src/Database/Adapter/Postgres.php [hunk 2]
```diff
@@ -346,9 +346,17 @@ public function deleteAttribute(string $collection, string $id, bool $array = fa
 
         $sql = $this->trigger(Database::EVENT_ATTRIBUTE_DELETE, $sql);
 
-        return $this->getPDO()
+        try {
+            return $this->getPDO()
             ->prepare($sql)
             ->execute();
+        } catch (PDOException $e) {
+            if ($e->getCode() === "4242703000" && $e->errorInfo[1] === 7) {
```
Comment (abnegate) [inline src/Database/Adapter/Postgres.php:354]: Is this really the code 😅

**Review by abnegate — COMMENTED** [review]
Let's add a test to make sure this works as expected

File: src/Database/Adapter/Postgres.php (hunk 2 above)
Comment (PineappleIOnic) [reply src/Database/Adapter/Postgres.php]: No, I don't know where I got that code. Maybe my keyboard messaged up 😅

File: tests/e2e/Adapter/MariaDBTest.php [hunk 3]
```diff
@@ -56,6 +57,17 @@ public static function getDatabase(bool $fresh = false): Database
 
         $database->create();
 
+        self::$pdo = $pdo;
         return self::$database = $database;
     }
+
+    protected static function deleteColumn(string $collection, string $column): bool
+    {
+        $SQLTable = "`" . self::getDatabase()->getDatabase() . "`.`" . self::getDatabase()->getNamespace() . "_" . $collection . "`";
```
Comment (abnegate) [inline tests/e2e/Adapter/MariaDBTest.php:66]: Let's use lower for SQL here `$sqlTable`

File: tests/e2e/Adapter/Base.php [hunk 4]
```diff
@@ -1295,6 +1303,11 @@ public function testCreateDeleteAttribute(): void
         } catch (Exception $e) {
             $this->assertInstanceOf(DuplicateException::class, $e);
         }
+
+        // Test delete attribute when column does not exist
+        $this->assertEquals(true, static::getDatabase()->createAttribute('attributes', 'string1', Database::VAR_STRING, 128, true));
+        $this->assertEquals(true, static::deleteColumn('attributes', 'string1'));
+        $this->assertEquals(true, static::getDatabase()->deleteAttribute('attributes', 'string1'));
```
Comment (abnegate) [inline tests/e2e/Adapter/Base.php:1310]: Let's also assert that it's remove from the collections metadata after this

File: tests/e2e/Adapter/MirrorTest.php [hunk 5]
```diff
@@ -22,6 +22,7 @@
 class MirrorTest extends Base
 {
     protected static ?Mirror $database = null;
+    protected static ?PDO $pdo = null;
```
Comment (abnegate) [inline tests/e2e/Adapter/MirrorTest.php:25]: Mirror should cache both source and destination PDO and delete the column from both

File: tests/e2e/Adapter/MirrorTest.php [hunk 6]
```diff
@@ -312,4 +314,14 @@ public function testDeleteMirroredDocument(): void
         $this->assertTrue($database->getSource()->getDocument('testDeleteMirroredDocument', $document->getId())->isEmpty());
         $this->assertTrue($database->getDestination()->getDocument('testDeleteMirroredDocument', $document->getId())->isEmpty());
     }
+
+    protected static function deleteColumn(string $collection, string $column): bool
+    {
+        $sqlTable = "`" . self::getDatabase()->getDatabase() . "`.`" . self::getDatabase()->getNamespace() . "_" . $collection . "`";
```
Comment (abnegate) [inline tests/e2e/Adapter/MirrorTest.php:320]: getDatabase and getNamespace should be called on source and destination DBs, not the mirror

**Review by abnegate — APPROVED** [review]

**Pull request merged**